book/src/safety/README.md

```diff
@@ -4,18 +4,22 @@
 
 ## Type Safety
 
-The code that `swift-bridge` is type safe on both the Rust and Swift side,
-so all of your interfacing between the two languages is type safe.
+All of the Rust and Swift FFI code that `swift-bridge` generates
+for you is type safe.
 
 ## Memory Safety
 
-Needing to worry about memory safety should be very uncommon when using `swift-bridge`.
+`swift-bridge` guarantees that the Rust side of your FFI boundary
+is memory safe.
```

Comment:
First let me try to explain what I think can be improved in the snippet above.

> the Rust side of your FFI boundary is memory safe

This seems to be a bit vague.
- One counter-example is that safe Rust can end up dereferencing an already freed memory (e.g. if Rust gets `&'static T` from Swift but Swift frees that).
- I guess that the sentence above tries to say that safe Rust is not the **root** cause of the UaF (e.g. tries to say that the Swift side of FFI is at fault).

> `swift-bridge` guarantees

This may be misunderstood / this seems to overpromise?

Maybe you would consider rewriting to something like:

-------------------

The memory safety of a Rust program depends on following a few rules in `unsafe` blocks.  A program using a mix of Swift and Rust needs to ensure that  Swift code follows similar rules.  Most of the rules are related to references:
- When Swift receives a reference from Rust, one needs to ensure that:
    -  Swift doesn't dereference the value outside of the reference lifetime
- When Swift passes a reference to Rust, one needs to ensure that:
    - Rust lifetime expectations are met
    - Mutable Rust references are exclusive (i.e. there are no other Rust references to the same memory)
    - Swift doesn't mutate data behind immutable Rust references

Reply:
Super clean!! Thanks!!

I'll apply your feedback pretty much as is.